core-bundle/src/InsertTag/InsertTagParser.php

```diff
@@ -342,6 +362,49 @@ private function doParse(string $input): ParsedSequence
         return new ParsedSequence($result);
     }
 
+    private function executeAsResponse(ParsedSequence|string $input, bool $allowEsiTags, OutputType $sourceType = OutputType::text): Response
+    {
+        $response = new Response();
+        $response->headers->set('Content-Type', OutputType::html === $sourceType ? 'text/html' : 'text/plain');
+
+        $replaced = $this->executeReplace($input, $allowEsiTags, $sourceType);
+
+        $expiresAt = null;
+        $cacheTags = [];
+        $contents = [];
+
+        foreach ($replaced as $result) {
+            if (OutputType::html === $result->getOutputType()) {
+                $contents[] = $result->getValue();
+            } else {
+                $contents[] = StringUtil::specialchars($result->getValue());
+            }
+
+            if ($result->getExpiresAt() && (!$expiresAt || $expiresAt->diff($result->getExpiresAt())->invert)) {
+                $expiresAt = $result->getExpiresAt();
+            }
+
+            foreach ($result->getCacheTags() as $cacheTag) {
+                $cacheTags[] = $cacheTag;
+            }
+        }
+
+        if ($expiresAt) {
+            $response->setSharedMaxAge($expiresAt->getTimestamp() - (new \DateTimeImmutable())->getTimestamp());
+        }
+
+        if ($cacheTags) {
```

Comment:
If we forward the call normally, we'll also have `kernel.response` listeners so all of this is not needed, is it?